### PR TITLE
titles: use rdrama's title finding code

### PR DESCRIPTION
**Pull request by justcool393**
this fixes a potential DoS in some really weird pages (seems to be a bug with BS4) 
we're not parsing arbitrary HTML and the title tag is probably one of the simplest tags so i think this is acceptable
in addition we make some nice checks

### CODE REVIEW

**Comment by justcool393** [comment]
reference: https://github.com/Aevann1/rDrama/commit/f5912bb4cc0dd3d0c7a88b32f077255e3b6dd186 (and a couple of commits that came after it but i forgot what they were)

**Comment by justcool393** [comment]
image

image

**Comment by TLSM** [comment]
The precise nature of the DoS was that a dependency of BS4 would get caught in an infinite loop doing charset detection on certain pages (most frequently some YouTube pages). We still have no idea why, even after investigating the dependency code itself. However, it was sufficient to crash the worker and max out a core for quite some time.

**Comment by justcool393** [comment]
apparently looking for meta charset tags or meta http-equiv tags is too hard for BS4 :pensive:

**Comment by justcool393** [comment]
regardless this should be faster anyway since we don't actually need to parse the HTML nor even really care about its contents other than the title. requests parses the HTTP thing from Content-Type or Content-Encoding headers so it shouldn't be an issue

and if it isn't well formed enough for this it's not worth caring about anyway since there are probably other major issues with the page being linked to

**Comment by TLSM** [comment]
Bug discovered on upstream, not yet integrated into this PR: need to run retrieved title through `html.unescape` before returning. Many sites include HTML character references in their `<title>`, which otherwise display literally when used as an `<input>` value—which is a safe context that doesn't need character references.

**Comment by justcool393** [comment]
fixed @TLSM

File: files/routes/posts.py [hunk 1]
```diff
@@ -1147,16 +1147,24 @@
...
 def get_post_title(v):
-
 	url = request.values.get("url")
-	if not url: abort(400)
+	if not url or '\\' in url: abort(400)
+	url = url.strip()
+	if not url.startswith('http'): abort(400)
+
+	checking_url = url.lower().split('?')[0].split('%3F')[0]
+	if any((checking_url.endswith(f'.{x}') for x in extensions)):
```
[the remark this answers is not in the document]
Comment (justcool393) [reply files/routes/posts.py]: image

**Review by TLSM — APPROVED** [review]
Faithful replica of operative and stable code on the upstream. Tested locally and works. Fixes an actual DoS.

**Review by zorbathut — APPROVED** [review]

**Pull request merged**